eZ/Publish/API/Repository/Values/Content/VersionDiff/VersionDiff.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Values\Content\VersionDiff;
+
+use eZ\Publish\API\Repository\Values\ValueObject;
+use OutOfBoundsException;
+
+class VersionDiff extends ValueObject
```

Comment:
DX: 

A) VerssionDiff should implement`IteratorAggregate`.  Currently, it's not possible to iterate over FieldDiffs without companion content type/content object.

```php
$versionDiff = $this->comparisonService->compareVersions($versionA, $versionB, $languageCode);

$contentType = $this->contentTypeSerivce->loadContentTypeByIdentifier('article');
foreach ($contentType->fieldDefinitions as $fieldDefinition) {
    $fieldDiff = $versionDiff->getFieldDiffByIdentifier($fieldDefinition->identifier);

    dump($fieldDiff->getFieldDef()->getName());
    dump($fieldDiff->getDiffValue());
}
```

VS 

```php
$versionDiff = $this->comparisonService->compareVersions($versionA, $versionB, $languageCode);

/** @var \eZ\Publish\API\Repository\Values\Content\VersionDiff\FieldDiff $fieldDiff */
foreach ($versionDiff as $fieldDiff) {
    dump($fieldDiff->getFieldDef()->getName());
    dump($fieldDiff->getDiffValue());
}
```

B) How to check if versions A is equal/not equal to version B?

Reply:
https://github.com/ezsystems/ezpublish-kernel/pull/2819/commits/fc3dd88c0c2c20193f241b2716fc2738dc6544dc

https://github.com/ezsystems/ezpublish-kernel/pull/2819/commits/2f7221d6993d8dfbd6f2c38e60c28395a3528ff9